eng/helix/content/RunTests/Program.cs

```diff
@@ -0,0 +1,180 @@
+            environmentVariables.Add("NUGET_RESTORE", nugetRestore);
+            var dotnetEFFullPath = Path.Combine(nugetRestore, $"dotnet-ef/{efVersion}/tools/netcoreapp3.1/any/dotnet-ef.exe");
+            Console.WriteLine($"Set DotNetEfFullPath: {dotnetEFFullPath}");
+            environmentVariables.Add("DotNetEfFullPath", dotnetEFFullPath);
+
+            Console.WriteLine("Checking for Microsoft.AspNetCore.App");
+            if (Directory.Exists("Microsoft.AspNetCore.App"))
+            {
+                Console.WriteLine($"Found Microsoft.AspNetCore.App, copying to {dotnetRoot}/shared/Microsoft.AspNetCore.App/{runtimeVersion}");
+                foreach (var file in Directory.EnumerateFiles("Microsoft.AspNetCore.App", "*.*", SearchOption.AllDirectories))
+                {
+                    File.Copy(file, $"{dotnetRoot}/shared/Microsoft.AspNetCore.App/{runtimeVersion}", overwrite: true);
+                }
+
+                Console.WriteLine($"Adding current directory to nuget sources: {HELIX_WORKITEM_ROOT}");
+
+                await ProcessUtil.RunAsync($"{dotnetRoot}/dotnet",
+                    $"nuget add source {HELIX_WORKITEM_ROOT} --configfile NuGet.config",
+                    environmentVariables: environmentVariables);
+
+                await ProcessUtil.RunAsync($"{dotnetRoot}/dotnet",
+                    "nuget add source https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet5/nuget/v3/index.json --configfile NuGet.config",
+                    environmentVariables: environmentVariables);
+
+                await ProcessUtil.RunAsync($"{dotnetRoot}/dotnet",
+                    "nuget list source",
+                    environmentVariables: environmentVariables);
+
+                await ProcessUtil.RunAsync($"{dotnetRoot}/dotnet",
+                    $"tool install dotnet-ef --global --version {efVersion}",
+                    environmentVariables: environmentVariables);
+
+                path += $";{Environment.GetEnvironmentVariable("DOTNET_CLI_HOME")}/.dotnet/tools";
+                environmentVariables["PATH"] = path;
+            }
+
+            Directory.CreateDirectory(nugetRestore);
+
+            // Rename default.runner.json to xunit.runner.json if there is not a custom one from the project
+            if (!File.Exists("xunit.runner.json"))
+            {
+                File.Copy("default.runner.json", "xunit.runner.json");
+            }
+
+            Console.WriteLine();
+            Console.WriteLine("Displaying directory contents");
+            foreach (var file in Directory.EnumerateFiles("./"))
+            {
+                Console.WriteLine(Path.GetFileName(file));
+            }
+            Console.WriteLine();
+
+            var discoveryResult = await ProcessUtil.RunAsync($"{dotnetRoot}/dotnet",
+                $"vstest {target} -lt",
+                environmentVariables: environmentVariables);
+
+            if (discoveryResult.StandardOutput.Contains("Exception thrown"))
```

Comment:
Why would this happen?

Reply:
Is it ever in stderr?

Reply:
I'm not familiar with the specifics of this
https://github.com/dotnet/aspnetcore/pull/8107